rosbag2_keyboard_handler/include/rosbag2_keyboard_handler/IKeyboardHandler.hpp

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 Apex.AI, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_KEYBOARD_HANDLER__IKEYBOARDHANDLER_HPP_
+#define ROSBAG2_KEYBOARD_HANDLER__IKEYBOARDHANDLER_HPP_
+
+#include <functional>
+#include <string>
+
+class IKeyboardHandler
```

Comment:
This is looking very organized, but I wonder if it's a little over-engineered. Is there any specific need for a separation of `IKeyboardHandler` and `KeyboadHandlerBase`?

Reply:
I come to it after establishing `IKeyboardHandler` as an interface and found out that some part is going to be common for Unix and Windows version and I moved it in `KeyboardHandlerBase`.